src/main/include/utils/StateManager.h

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include <frc2/command/CommandPtr.h>
+#include <frc2/command/CommandScheduler.h>
+#include <frc2/command/DeferredCommand.h>
+#include <frc2/command/InstantCommand.h>
+#include <frc2/command/SubsystemBase.h>
+
+template <typename StateType>
```

Comment:
You can change `StateType` to just `T`

Reply:
Isn’t that a loss of descriptiveness?

Reply:
Eh, not for generics like this where it's just one. Plus you can add a doxygen comment above the typename declaration with something like:

```
/**
 * @tparam T The key type for setting and accessing commands
*/
```